### PR TITLE
Remove xstructhash

### LINKED ISSUE #1093: Use Hash.Dump instead of xstructhash or structhash
- [ ] Function that need to hash a struct should use `Hash.Dump` instead of `xstructhash` or `structhash` (but keep `structhash` under the hood)
- [ ] `xstructhash` should be removed
- [ ] Also, it would make sense that the `Dump` function return `Digest` instead of `Hash`, so more data can be added to the `Digest`. @krhubert what do you think?

### PR DESCRIPTION
**Pull request by krhubert**
close #1093

### CODE REVIEW

File: sdk/instance/helpers.go [hunk 1]
```diff
@@ -77,9 +76,9 @@ func extractVolumesFrom(s *service.Service, d *service.Dependency) ([]container.
...
-	return hex.EncodeToString(xstructhash.Hash([]string{
-		s.Sid,
-		dependency,
-		volume,
-	}, 1))
+	d := sha1.New()
+	d.Write([]byte(s.Sid))
+	d.Write([]byte(dependency))
+	d.Write([]byte(volume))
+	return hex.EncodeToString(d.Sum(nil))
```
Comment (NicolasMahe) [inline sdk/instance/helpers.go]: Did you try to use Hash.Hash?
A sha256 base58 encoded should be small enough for docker?

File: sdk/service/service.go [hunk 2]
```diff
@@ -57,7 +56,7 @@ func (s *Service) Create(srv *service.Service) (*service.Service, error) {
 
 	// calculate and apply hash to service.
 	dh := dirhash.New(path)
```
Comment (NicolasMahe) [inline sdk/service/service.go]: not blocking: could you also move dirhash in Hash package?

File: sdk/instance/helpers.go (hunk 1 above)
Comment (krhubert) [reply sdk/instance/helpers.go]: No, I have not tested, but I can try. I thought it would be too much.

Comment (krhubert) [reply sdk/instance/helpers.go]:
```
        "Mounts": [
            {
                "Type": "volume",
                "Name": "8RBsoeyoRwajj86MZfZE6gMDJQVYGYcdSfx1zxqxNHbr",
                "Source": "/var/lib/docker/volumes/8RBsoeyoRwajj86MZfZE6gMDJQVYGYcdSfx1zxqxNHbr/_data",
```
It's working

**Pull request merged**